src/ssl.c

```diff
@@ -17813,24 +17835,62 @@ size_t wolfSSL_get_client_random(const WOLFSSL* ssl, unsigned char* out,
 #endif
     }
 
+    /* If a valid struct is provided with function pointers, will override
+       RAND_seed, bytes, cleanup, add, pseudo_bytes and status.  If a NULL
+       pointer is passed in, it will cancel any previous function overrides.
+
+       Returns WOLFSSL_SUCCESS on success, WOLFSSL_FAILURE on failure. */
+    int wolfSSL_RAND_set_rand_method(const WOLFSSL_RAND_METHOD *methods)
+    {
+        int ret = WOLFSSL_FAILURE;
+
+        if (wc_LockMutex(&rand_methods_mutex) != 0) {
+            WOLFSSL_MSG("Lock rand methods mutex failed");
+            ret = BAD_MUTEX_E;
+        }
+        else {
+            rand_methods = methods;
+            wc_UnLockMutex(&rand_methods_mutex);
+            ret = WOLFSSL_SUCCESS;
+        }
+
+        return ret;
+    }
 
     int wolfSSL_RAND_status(void)
     {
-        return WOLFSSL_SUCCESS;  /* wolfCrypt provides enough seed internally */
+        int ret = WOLFSSL_FAILURE;
+
+        if (wc_LockMutex(&rand_methods_mutex) != 0) {
+            WOLFSSL_MSG("Lock rand methods mutex failed");
+            ret = BAD_MUTEX_E;
```

Comment:
If the lock fails you need to return error here and not call unlock... Same for others below.